engines/python/setup/djl_python/deepspeed.py

```diff
@@ -10,98 +10,246 @@
+    "gpt-neox": {transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXLayer: GPTNEOXLayerPolicy},
+    "bloom": {transformers.models.bloom.modeling_bloom.BloomBlock: BLOOMLayerPolicy},
+}
 
 
 class DeepSpeedService(object):
 
     def __init__(self):
-        self.predictor = None
-        self.max_new_tokens = 0
+        self.pipeline = None
         self.initialized = False
+        self.ds_config = None
+        self.task = None
+        self.logger = logging.getLogger()
+        self.model_dir = None
+        self.model_id = None
+        self.data_type = None
+        self.max_tokens = None
+        self.device = None
+        self.world_size = None
+        self.tensor_parallel_degree = None
+        self.model_config = None
 
     def initialize(self, properties: dict):
-        self.max_new_tokens = int(properties.get("max_new_tokens", "50"))
-        model_dir = properties.get("model_dir")
-        data_type = properties.get("data_type", "fp32")
-        mp_size = int(properties.get("tensor_parallel_degree", "1"))
-        model_id = properties.get("model_id")
-        # LOCAL_RANK env is initialized after constructor
-        device = int(os.getenv('LOCAL_RANK', '0'))
-        if not model_id:
-            model_id = model_dir
-            config_file = os.path.join(model_dir, "config.json")
+        self.parse_properties(properties)
+        self.validate_model_type_and_task()
+        self.create_model_pipeline()
+        self.logger.info(f"Initialized DeepSpeed model with the following configurations"
+                         f"model: {self.model_id}"
+                         f"task: {self.task}"
+                         f"data_type: {self.data_type}"
+                         f"tensor_parallel_degree: {self.tensor_parallel_degree}")
+        self.initialized = True
+
+    def parse_properties(self, properties):
+        self.model_dir = properties.get("model_dir")
+        self.model_id = properties.get("model_id")
+        self.task = properties.get("task")
+        self.data_type = properties.get("data_type", "fp32")
+        self.max_tokens = int(properties.get("max_tokens", 1024))
+        self.device = int(os.getenv("LOCAL_RANK", 0))
+        self.world_size = int(os.getenv("WORLD_SIZE", 1))
+        self.tensor_parallel_degree = int(properties.get("tensor_parallel_degree", self.world_size))
+        self.ds_config = {
+            "replace_with_kernel_inject": True,
+            "dtype": self.data_type,
+            "tensor_parallel": {
+                "enabled": True,
+                "tp_size": self.tensor_parallel_degree,
+                "mpu": None,
+            },
+            "enable_cuda_graph": properties.get("enable_cuda_graph", "false").lower() == "true",
+            "triangular_masking": properties.get("triangular_masking", "true").lower() == "true",
+            "checkpoint": properties.get("checkpoint"),
+            "base_dir": properties.get("base_dir"),
```

Comment:
Why user need to set `base_dir`?

Reply:
If base_dir is not set, DS will look for checkpoint in current directory. If checkpoint is in some child dir then base_dir helps to identify it (though it works if you specify the full path to checkpoint and leave base_dir empty). Not sure if we need to expose this, but if user is coming from DS world they might expect this.